torch/csrc/jit/codegen/cuda/runtime/memory.cu

```diff
@@ -0,0 +1,105 @@
+// Utility macro for this file
+#define DEVICE_INLINE __device__ inline
+
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 750))
+
+namespace Turing {
+
+namespace util {
+
+// Special utility for ldmatrix and cp_async
+DEVICE_INLINE unsigned toSmem(const void* ptr) {
+  unsigned smem_ptr;
+
+  asm("{ .reg .u64 smem_ptr; cvta.to.shared.u64 smem_ptr, %1; cvt.u32.u64 %0, smem_ptr; }\n"
+      : "=r"(smem_ptr)
+      : "l"(ptr));
+
+  return smem_ptr;
+}
+
+} // namespace util
+
+DEVICE_INLINE void ldMatrix(Array<__half, 4>& out, void const* ptr) {
```

Comment:
@rdspring1 FYI, Turing/Ampere mma's only have TN but in the case of fp16, ldmatrix can transpose it. 

So we'd need to handle transpose or transposed layout for types other than fp16 input, i.e. double or tf32.